polynote-kernel/src/main/scala/polynote/kernel/remote/transport.scala

```diff
@@ -401,19 +413,61 @@ object SocketTransport {
       * Deploy by starting a Java process that inherits classpath and environment variables from this process
       */
     class DeployJava[KernelFactory <: Kernel.Factory.Service : ClassTag] extends DeployCommand {
-      override def apply(serverAddress: InetSocketAddress, classPath: Seq[Path]): RIO[Config with CurrentNotebook, Seq[String]] = ZIO {
-        val java = Paths.get(System.getProperty("java.home"), "bin", "java").toString
-        val javaArgs = sys.process.javaVmArguments.filterNot(_ startsWith "-agentlib")
-        val fullClassPath = (classPath.map(_.toString) :+ System.getProperty("java.class.path"))
-          .filterNot(_.isEmpty)
-          .mkString(File.pathSeparator)
-
-        java :: "-cp" :: fullClassPath :: javaArgs :::
-          classOf[RemoteKernelClient].getName ::
-          "--address" :: serverAddress.getAddress.getHostAddress ::
-          "--port" :: serverAddress.getPort.toString ::
-          "--kernelFactory" :: classTag[KernelFactory].runtimeClass.getName ::
-          Nil
+      private def findJava: URIO[BaseEnv, String] =
+        property("java.home").mapError(_.getMessage).someOrFail("No java.home property is set")
+          .map(home => Paths.get(home, "bin", "java").toString)
+          .tapError(err => Logging.warn("Couldn't find java executable; will just use 'java' ($err)"))
+          .orElse(ZIO.succeed("java"))
+
+      // parse a JVM args string into a list of args
+      private def parseJVMArgs(str: String): List[String] = {
```

Comment:
Is this parsing just turning `"-Dfoo=bar -Dbar=baz -Xmx200m"` into `List("-Dfoo=bar",  "-Dbar=baz",  "-Xmx200m)"` or is there more to it?

Reply:
Also, what do you think about adding some tests to this parser? 😉 

idk, I see strings and regexes and I think "tests" 😛